modules/improvedigitalBidAdapter.js

```diff
@@ -192,539 +265,454 @@ export const spec = {
+      if (nativeItem) {
+        const asset = {
+          id: nativeItem.id,
+          required: cur.required ? 1 : 0,
+        };
+        switch (nativeItem.assetType) {
+          case NATIVE_DATA.ASSET_TYPES.TITLE:
+            asset.title = {len: cur.len || nativeItem.default.len};
+            break;
+          case NATIVE_DATA.ASSET_TYPES.DATA:
+            asset.data = cleanObj({type: nativeItem.type, len: cur.len})
+            break;
+          case NATIVE_DATA.ASSET_TYPES.IMG:
+            const img = {
+              type: nativeItem.type
+            }
+            if (cur.sizes) {
+              [img.w, img.h] = cur.sizes;
+            } else if (cur.aspect_ratios) {
+              img.wmin = cur.aspect_ratios[0].min_width;
+              img.hmin = cur.aspect_ratios[0].min_height;
+            }
+            asset.img = cleanObj(img);
+            break;
+          default:
+            return;
+        }
+        request.assets.push(asset);
+      }
     }
-    if (placementObject.adUnitId) {
-      outputObject.adUnitId = placementObject.adUnitId;
+    return { ver: NATIVE_DATA.VERSION, request: JSON.stringify(request) };
+  },
+};
+
+const ID_RESPONSE = {
+  buildAd(bidObject, bid, bidRequest, bidResponse) {
+    if (deepAccess(bidObject, 'mediaTypes.video')) {
+      ID_RESPONSE.buildVideoAd(bid, bidRequest, bidResponse);
+    } else if (deepAccess(bidObject, 'mediaTypes.banner')) {
+      ID_RESPONSE.buildBannerAd(bid, bidRequest, bidResponse);
+    } else if (deepAccess(bidObject, 'mediaTypes.native')) {
+      ID_RESPONSE.buildNativeAd(bid, bidRequest, bidResponse)
     }
-    if (placementObject.currency) {
-      impressionObject.currency = placementObject.currency.toUpperCase();
+  },
+
+  buildVideoAd(bid, bidRequest, bidResponse) {
+    bid.mediaType = VIDEO;
+    bid.vastXml = bidResponse.adm;
+    if (ID_UTILITY.isOutstreamVideo(bidRequest)) {
+      bid.adResponse = {
+        content: bid.vastXml,
+        height: bidResponse.h,
+        width: bidResponse.w,
+      }
+      bid.renderer = ID_UTILITY.createRenderer(bidRequest);
     }
-    if (placementObject.bidFloor) {
-      impressionObject.bidfloor = placementObject.bidFloor;
+  },
+
+  buildBannerAd(bid, bidRequest, bidResponse) {
+    bid.mediaType = BANNER;
+    if (bidResponse.nurl) {
+      bid.nurl = bidResponse.nurl;
     }
-    if (placementObject.bidFloorCur) {
-      impressionObject.bidfloorcur = placementObject.bidFloorCur.toUpperCase();
+    if (bidResponse.adm) {
+      bid.ad = bidResponse.adm;
+      bid.width = bidResponse.w;
+      bid.height = bidResponse.h;
     }
-    if (placementObject.placementId) {
-      impressionObject.pid = placementObject.placementId;
+  },
+
+  buildNativeAd(bid, bidRequest, bidResponse) {
+    bid.mediaType = NATIVE;
+    const native = JSON.parse(bidResponse.adm)
+    const nativeAd = {
+      clickUrl: native.link.url,
+      impressionTrackers: native.imptrackers,
```

Comment:
As discussed on the call, the logic should be for trackers like in the old adapter: https://github.com/prebid/Prebid.js/blob/master/modules/improvedigitalBidAdapter.js#L445